### PR TITLE
Add py3 native image

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
Ready

## Related Content Pull Request
Related PR: link to the PR at demisto/content

## Related Issues
[link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-4374)

## Description
add py3 master image

### CODE REVIEW

File: docker/py3-primary/chromium/download_chromedriver.sh
Comment (JudahSchwartz) [inline docker/py3-primary/chromium/download_chromedriver.sh:7]: wdym installed chrome version? Can we use this for chromium image too?

File: docker/py3-primary/Dockerfile
Comment (JudahSchwartz) [inline docker/py3-primary/Dockerfile:19]: google-chrome-stable will take from google-chrome in the repo folder?

File: docker/py3-primary/office-utils/LibreOffice_7.4.2.3_Linux_x86-64_rpm/readmes/README_en-US
Comment (GuyAfik) [reply docker/py3-primary/office-utils/LibreOffice_7.4.2.3_Linux_x86-64_rpm/readmes/README_en-US:3]: it was inside the tar, can be removed

File: docker/py3-primary/office-utils/LibreOffice_7.4.2.3_Linux_x86-64_rpm/install
Comment (GuyAfik) [reply docker/py3-primary/office-utils/LibreOffice_7.4.2.3_Linux_x86-64_rpm/install:3]: this is the result of the zip extraction from the official libreoffice site: https://www.libreoffice.org/donate/dl/rpm-x86_64/7.4.2/en-US/LibreOffice_7.4.2_Linux_x86-64_rpm.tar.gz - i was unable to untar it inside the container

File: docker/py3-primary/chromium/download_chromedriver.sh
Comment (GuyAfik) [reply docker/py3-primary/chromium/download_chromedriver.sh:7]: its taken directly from chromium image, its the same script.

it downloads the corresponding chromedriver according to the google-version installed

File: docker/py3-primary/Dockerfile
Comment (GuyAfik) [reply docker/py3-primary/Dockerfile:19]: exactly

File: docker/py3-primary/pyproject.toml
Comment (GuyAfik) [reply docker/py3-primary/pyproject.toml:64]: dev-bug that its purpose is..?

File: docker/py3-primary/verify.py
Comment (GuyAfik) [reply docker/py3-primary/verify.py:2]: 1. not all the images have verify.py, some of them are missing (like tesseract for example)
2. i believe this is a scope of another dev-task - this shouldn't be hard though

File: docker/py3-primary/Dockerfile
Comment (GuyAfik) [reply docker/py3-primary/Dockerfile:65]: i agree - this is just for now for making stuff clear during development

File: .github/dependabot.yml
Comment (GuyAfik) [reply .github/dependabot.yml:792]: i don't find such option

File: docker/build_docker.sh
Comment (GuyAfik) [reply docker/build_docker.sh:179]: do you mean to add it to the `dockerbuild` image? - https://github.com/demisto/dockerfiles/tree/master/docker/dockerbuild

File: .github/dependabot.yml
Comment (GuyAfik) [reply .github/dependabot.yml:792]: just found it, will update to poetry https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem

File: docker/py3-primary/chromium/download_chromedriver.sh
Comment (tkatzir) [inline docker/py3-primary/chromium/download_chromedriver.sh:19]:
```suggestion
    echo "Something went wrong with getting chrome version. Aborting!!!"
```

File: docker/py3-primary/chromium/download_chromedriver.sh
Comment (tkatzir) [inline docker/py3-primary/chromium/download_chromedriver.sh:23]:
```suggestion
echo "Using GOOGLE_CHROME_VERSION: $GOOGLE_CHROME_VERSION"
```

File: docker/py3-primary/chromium/download_chromedriver.sh
Comment (tkatzir) [inline docker/py3-primary/chromium/download_chromedriver.sh:25]:
```suggestion
echo "Downloading chromedriver list..."
```

File: docker/py3-primary/chromium/download_chromedriver.sh
Comment (tkatzir) [inline docker/py3-primary/chromium/download_chromedriver.sh:30]:
```suggestion
echo "Using chromedriver version: $DRIVER_VERSION"
```

**Review by tkatzir — COMMENTED** [review]
Looks great!
Can we store the binary files somewhere outside the source tree?

File: .github/dependabot.yml
Comment (GuyAfik) [reply .github/dependabot.yml:792]: getting this error: `The property '#/updates/177/package-ecosystem' value "poetry" did not match one of the following values: npm, bundler, composer, maven, mix, cargo, gradle, nuget, gomod, docker, elm, gitsubmodule, github-actions, pip, terraform, pub`

File: docker/py3-primary/office-utils/LibreOffice_7.4.2.3_Linux_x86-64_rpm/install
Comment (GuyAfik) [reply docker/py3-primary/office-utils/LibreOffice_7.4.2.3_Linux_x86-64_rpm/install]: removed

File: docker/py3-primary/Dockerfile
Comment (GuyAfik) [reply docker/py3-primary/Dockerfile]: done

**Comment by GuyAfik** [comment]
> Looks great! Can we store the binary files somewhere outside the source tree?

removed the binary files, installing libreoffice with wget now

File: docker/build_docker.sh
Comment (GuyAfik) [reply docker/build_docker.sh:179]: @tkatzir @JudahSchwartz please refer this in py3-tools PR as well - https://github.com/demisto/dockerfiles/pull/9337

File: docker/py3-native/Dockerfile
Comment (tkatzir) [inline docker/py3-native/Dockerfile:7]: Are all these comments needed?

**Review by tkatzir — COMMENTED** [review]
Looks great, one question.

File: docker/build_docker.sh
Comment (JudahSchwartz) [reply docker/build_docker.sh:179]: @GuyAfik yes

Comment (GuyAfik) [reply docker/build_docker.sh:179]: https://github.com/demisto/dockerfiles/pull/10296

File: docker/build_docker.sh
Comment (JudahSchwartz) [inline docker/build_docker.sh:182]: move this to dockerbuild also.
We should probably run `poetry env remove` here like we do for pipenv

File: .github/dependabot.yml
Comment (JudahSchwartz) [reply .github/dependabot.yml:792]: yeah I think it should be pip for poetry, acc to their docs

File: docker/build_docker.sh
Comment (GuyAfik) [reply docker/build_docker.sh:182]: done, once https://github.com/demisto/dockerfiles/pull/10296 and https://github.com/demisto/dockerfiles/pull/9337 are merged ill remove this part.

File: docker/py3-native/Dockerfile
Comment (GuyAfik) [reply docker/py3-native/Dockerfile:7]: no they can be removed before merge, these are just for making things easier for me when to follow up.

File: docker/py3-native/Dockerfile
Comment (tkatzir) [inline docker/py3-native/Dockerfile:80]: Any reason why this is not part of the previous `RUN` command (that started on line 9) ?

Comment (GuyAfik) [reply docker/py3-native/Dockerfile:80]: no particular reason, it just to differentiate between chromium + tesseract which are "special" images and more "complex" images than others. 

The tesseract and chromium require more OS dependancies than "regular" docker images. 

I think splitting "complex" docker images (which require a lot of os dependancies) - will make it easier to read in the future. 

Shoving all in one RUN command would make it probably less readable

File: docker/py3-native/verify.py
Comment (JudahSchwartz) [inline docker/py3-native/verify.py:64]: can we import these tests from where they sit in the repo instead of copying them to here?

File: docker/py3-native/Dockerfile
Comment (JudahSchwartz) [inline docker/py3-native/Dockerfile:1]: please add the last updated line like we have in other dockerfiles

File: docker/py3-native/Dockerfile
Comment (tkatzir) [reply docker/py3-native/Dockerfile:80]: Each RUN command adds a layer -> Larger image size.

Comment (GuyAfik) [reply docker/py3-native/Dockerfile:57]: I know, but for server it does not matter that much, also, the differences aren't so extreme.

File: docker/py3-native/Dockerfile
Comment (GuyAfik) [reply docker/py3-native/Dockerfile:1]: we don't have this in all other docker files, why do we need it? git can tell you the last time it was changed

Comment (JudahSchwartz) [reply docker/py3-native/Dockerfile:1]: iits useful for when we want to regenerate the image but we dont have any real changes to make to the file. The build wont catch it without some change

Comment (GuyAfik) [reply docker/py3-native/Dockerfile:1]: do i need to add it manually?

Comment (JudahSchwartz) [reply docker/py3-native/Dockerfile:1]: think youre right its probably not that important

File: docker/py3-native/verify.py
Comment (GuyAfik) [reply docker/py3-native/verify.py:64]: That is a bit problematic, what if someone adds a new dependency to image X, but in the native image we still do not support it? the `verify.py` will fail. If you think that this is good (and then we will know that this dependency needs to be added) i can do it with imports

Comment (GuyAfik) [reply docker/py3-native/verify.py:64]: also, i will need to copy the `verify.py` files to the container, or make it with a symbolic link, however symbolic link would work only a single file and not on all of them. What you want to achieve here is a bit complicated.

File: docker/py3-native/Dockerfile
Comment (GuyAfik) [reply docker/py3-native/Dockerfile:57]: does it matter so much that we will have 2 RUN commands instead of one? do you think it would make the image significantly smaller image?

File: docker/py3-native/verify.py
Comment (JudahSchwartz) [reply docker/py3-native/verify.py:64]: In the case you described, this either will be a reminder to update the master image as well, or we should drop support for that image if the tests dont pass and we dont want to support it
Cant we copy the files in in the same place we do verify.py?
@tkatzir  what do you think?

If this is too high effort for now lets open a task to do it after

Comment (JudahSchwartz) [reply docker/py3-native/verify.py:64]: we will need to touch build code so the docker has the files

File: docker/py3-native/Dockerfile
Comment (tkatzir) [reply docker/py3-native/Dockerfile:57]: You can try.
We can also try and squash layers during the build.

File: docker/py3-native/verify.py
Comment (GuyAfik) [reply docker/py3-native/verify.py]: I think we can do it with symlinks - testing it now. see the latest commit, if that works we can do a symbolic link to each docker image we would like.

File: docker/py3-native/Dockerfile
Comment (GuyAfik) [reply docker/py3-native/Dockerfile:57]: we try to do it in the next version, i don't think its urgent for the upcoming version, it would require me to test everything out again to see i didn't break something

File: docker/py3-native/verify.py
Comment (GuyAfik) [reply docker/py3-native/verify.py]: looks like it works, i updated the docker build a bit to run multiple `verify.py` files

File: docker/build_docker.sh
Comment (JudahSchwartz) [inline docker/build_docker.sh:237]: We're finding every verify py in the repo?

Comment (GuyAfik) [reply docker/build_docker.sh:237]: no, we are finding every `verify.py` for the entire folder which the docker resides in. in our case it will find all the files that their name is anything_verify.py and that are inside py3-native folder and will run them on the container.

**Comment by GuyAfik** [comment]
@tkatzir @JudahSchwartz please note I have added a short `README.md` as well.

File: docker/py3-primary/verify.py
Comment (JudahSchwartz) [reply docker/py3-primary/verify.py]: we implemented this in the end using symlinks

**Review by JudahSchwartz — APPROVED** [review]
What a beast

**Comment by JudahSchwartz** [comment]
lets make sure we get an approval from @glicht before merge

File: docker/py3-native/Dockerfile
Comment (glicht) [inline docker/py3-native/Dockerfile:28]: I think we should use a tag here to install and not from master. Probably use the latest tag of 1.82.0

Comment (GuyAfik) [reply docker/py3-native/Dockerfile:28]: agreed, will update it tomorrow

File: docker/py3-native/Dockerfile
Comment (GuyAfik) [reply docker/py3-native/Dockerfile:32]: agreed, will update it tomorrow

**Pull request merged**
